### ISSUE TITLE
Remove special case for `BindingPattern` in `GetSpan` codegen

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#3904]
As discussed in https://github.com/oxc-project/oxc/issues/3900#issuecomment-2188460531, it'd be preferable in my view to remove the "special case" code for `BindingPattern` from `GetSpan` codegen.

I propose instead adding a `#[span]` attr to the `kind` field, and codegen can act on that to generate:

```rs
impl<'a> GetSpan for BindingPattern<'a> {
    #[inline]
    fn span(&self) -> Span {
        self.kind.span()
    }
}
```

We can then use that same `#[span]` mechanism if it's required for any other types in future.

**Comment by Boshen** [comment]
Is this addressed?

**Comment by rzvxa** [comment]
@Boshen Aye, Captain!

Closed via this special case: https://github.com/oxc-project/oxc/blob/09f4d3c4d353222b60b8f6029136aefd41cb9677/tasks/ast_codegen/src/generators/impl_get_span.rs#L18

**Comment by rzvxa** [comment]
@Boshen Sorry, I thought it was the issue about removing the span from binding patterns.
------------

It will be addressed later on. We have all the infrastructure to make it happen. But since it currently works as is, I'm more focused on getting visitors and ast builders in before achieving this generator hint.

**Comment by rzvxa** [comment]
closed via #4012
